server/src/main/java/io/druid/segment/realtime/appenderator/BatchAppenderatorDriver.java

```diff
@@ -132,10 +132,11 @@ private SegmentsAndMetadata pushAndClear(
         .map(SegmentWithState::getSegmentIdentifier)
         .collect(Collectors.toList());
 
-    final ListenableFuture<SegmentsAndMetadata> future = Futures.transform(
-        pushInBackground(null, segmentIdentifierList),
-        this::dropInBackground
-    );
+    final ListenableFuture<SegmentsAndMetadata> future = ListenableFutures
+        .transformAsync(
```

Comment:
this line break doesn't seem necessary. It is a small thing but I still feel compelled to mention it.

Reply:
fixing